yarn-project/noir-compiler/src/compile/nargo.ts

```diff
@@ -35,20 +36,37 @@ export class NargoContractCompiler {
     const stdio = this.opts.quiet ? 'ignore' : 'inherit';
     const nargoBin = this.opts.nargoBin ?? 'nargo';
     const version = execSync(`${nargoBin} --version`, { cwd: this.projectPath, stdio: 'pipe' }).toString();
-    this.checkNargoBinVersion(version.replace('\n', ''));
+    this.checkNargoBinVersion(version);
     emptyDirSync(this.getTargetFolder());
     execSync(`${nargoBin} compile --no-backend`, { cwd: this.projectPath, stdio });
     return Promise.resolve(this.collectArtifacts());
   }
 
   private checkNargoBinVersion(version: string) {
-    if (!version.includes(NoirCommit)) {
-      this.log(
-        `Warning: the nargo version installed locally does not match the expected one. This may cause issues when compiling or deploying contracts. Consider updating your nargo or aztec-cli installation. \n- Expected: ${NoirTag} (git version hash: ${NoirCommit})\n- Found: ${version}`,
-      );
-    } else if (!this.opts.quiet) {
-      this.log(`Using ${version}`);
+    const outputLines = version.split('\n');
+    const nargoVersionLine = outputLines.find(line => line.indexOf('nargo') !== -1);
+    const nargoVer = nargoVersionLine?.match(/(\d+\.\d+\.\d+)/)?.[1];
+
+    if (!nargoVer) {
+      this.log('Warning: nargo version could not be determined.');
+      return;
+    }
+
+    const noirWasmSemver = parse(noirWasmVersion);
+
+    if (noirWasmSemver?.compareMain(nargoVer) === 0) {
```

Comment:
What does `compareMain do? Couldn't find it in the semver docs. Is it the same as `compare`?

Reply:
It ignores the pre-release information (ie. `-a098541.aztec`) otherwise 0.18.0 > 0.18.0-a098541.aztec